### ISSUE TITLE
[ASAN] Failure in mkfit

### ISSUE THREAD

**Issue by smuzaffar** [cms-sw/cmssw#37096]
Hi, 
  In ASAN IBs, many workflow are failing. ASAN complains about ( when build with `USSR_CXXFLAGS=-g`) 
```
==3127==ERROR: AddressSanitizer: invalid alignment requested in aligned_alloc: 64, alignment must be a power of two and the requested size 0x68 must be a multiple of alignment (thread T0)
    #0 0x7fcb2d092faf in __interceptor_aligned_alloc ../../../../libsanitizer/asan/asan_malloc_linux.cpp:192
    #1 0x7fcadb735c17 in mkfit::Pool<mkfit::CandCloner>::create() /build/muz/asan/CMSSW_12_3_ASAN_X_2022-02-28-1100/src/RecoTracker/MkFitCore/src/Pool.h:41
    #2 0x7fcadb735c17 in mkfit::Pool<mkfit::CandCloner>::populate(int) /build/muz/asan/CMSSW_12_3_ASAN_X_2022-02-28-1100/src/RecoTracker/MkFitCore/src/Pool.h:27
    #3 0x7fcadb735c17 in mkfit::ExecutionContext::populate(int) /build/muz/asan/CMSSW_12_3_ASAN_X_2022-02-28-1100/src/RecoTracker/MkFitCore/src/MkBuilder.cc:46
    #4 0x7fcadb735c17 in mkfit::MkBuilder::populate() /build/muz/asan/CMSSW_12_3_ASAN_X_2022-02-28-1100/src/RecoTracker/MkFitCore/src/MkBuilder.cc:171
    #5 0x7fcadb9f3920 in MkFitProducer::MkFitProducer(edm::ParameterSet const&) (/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_3_ASAN_X_2022-02-28-1100/lib/slc7_amd64_gcc11/pluginRecoTrackerMkFitPlugins.so+0x15a920)
    #6 0x7fcadba0ecc4 in edm::WorkerMaker<MkFitProducer>::makeModule(edm::ParameterSet const&) const (/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_3_ASAN_X_2022-02-28-1100/lib/slc7_amd64_gcc11/pluginRecoTrackerMkFitPlugins.so+0x175cc4)
    #7 0x7fcb2cda922d in edm::Maker::makeModule(edm::MakeModuleParams const&, edm::signalslot::Signal<void (edm::ModuleDescription const&)>&, edm::signalslot::Signal<void (edm::ModuleDescription const&)>&) const (/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_3_ASAN_X_2022-02-28-1100/lib/slc7_amd64_gcc11/libFWCoreFramework.so+0x82f22d)
    #8 0x7fcb2ca1a116 in edm::Factory::makeModule(edm::MakeModuleParams const&, edm::signalslot::Signal<void (edm::ModuleDescription const&)>&, edm::signalslot::Signal<void (edm::ModuleDescription const&)>&) const (/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_3_ASAN_X_2022-02-28-1100/lib/slc7_amd64_gcc11/libFWCoreFramework.so+0x4a0116)
    #9 0x7fcb2ca6de09 in edm::ModuleRegistry::getModule(edm::MakeModuleParams const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, edm::signalslot::Signal<void (edm::ModuleDescription const&)>&, edm::signalslot::Signal<void (edm::ModuleDescription const&)>&) (/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_3_ASAN_X_2022-02-28-1100/lib/slc7_amd64_gcc11/libFWCoreFramework.so+0x4f3e09)
    #10 0x7fcb2cdb6e31 in edm::WorkerRegistry::getWorker(edm::WorkerParams const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) (/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_3_ASAN_X_2022-02-28-1100/lib/slc7_amd64_gcc11/libFWCoreFramework.so+0x83ce31)
    #11 0x7fcb2cdab3f6 in edm::WorkerManager::getWorker(edm::ParameterSet&, edm::ProductRegistry&, edm::PreallocationConfiguration const*, std::shared_ptr<edm::ProcessConfiguration const>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) (/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_3_ASAN_X_2022-02-28-1100/lib/slc7_amd64_gcc11/libFWCoreFramework.so+0x8313f6)
    #12 0x7fcb2cdafd98 in edm::WorkerManager::addToUnscheduledWorkers(edm::ParameterSet&, edm::ProductRegistry&, edm::PreallocationConfiguration const*, std::shared_ptr<edm::ProcessConfiguration>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::set<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&) (/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_3_ASAN_X_2022-02-28-1100/lib/slc7_amd64_gcc11/libFWCoreFramework.so+0x835d98)
    #13 0x7fcb2cd0d4f1 in edm::StreamSchedule::StreamSchedule(std::shared_ptr<edm::TriggerResultInserter>, std::vector<edm::propagate_const<std::shared_ptr<edm::PathStatusInserter> >, std::allocator<edm::propagate_const<std::shared_ptr<edm::PathStatusInserter> > > >&, std::vector<edm::propagate_const<std::shared_ptr<edm::EndPathStatusInserter> >, std::allocator<edm::propagate_const<std::shared_ptr<edm::EndPathStatusInserter> > > >&, std::shared_ptr<edm::ModuleRegistry>, edm::ParameterSet&, edm::service::TriggerNamesService const&, edm::PreallocationConfiguration const&, edm::ProductRegistry&, edm::BranchIDListHelper&, edm::ExceptionToActionTable const&, std::shared_ptr<edm::ActivityRegistry>, std::shared_ptr<edm::ProcessConfiguration>, edm::StreamID, edm::ProcessContext const*) (/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_3_ASAN_X_2022-02-28-1100/lib/slc7_amd64_gcc11/libFWCoreFramework.so+0x7934f1)
    #14 0x7fcb2cc88eaa in edm::Schedule::Schedule(edm::ParameterSet&, edm::service::TriggerNamesService const&, edm::ProductRegistry&, edm::BranchIDListHelper&, edm::ProcessBlockHelperBase&, edm::ThinnedAssociationsHelper&, edm::SubProcessParentageHelper const*, edm::ExceptionToActionTable const&, std::shared_ptr<edm::ActivityRegistry>, std::shared_ptr<edm::ProcessConfiguration>, bool, edm::PreallocationConfiguration const&, edm::ProcessContext const*) (/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_3_ASAN_X_2022-02-28-1100/lib/slc7_amd64_gcc11/libFWCoreFramework.so+0x70eeaa)
    #15 0x7fcb2ccc492e in edm::ScheduleItems::initSchedule(edm::ParameterSet&, bool, edm::PreallocationConfiguration const&, edm::ProcessContext const*, edm::ProcessBlockHelperBase&) (/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_3_ASAN_X_2022-02-28-1100/lib/slc7_amd64_gcc11/libFWCoreFramework.so+0x74a92e)
    #16 0x7fcb2c8264de in edm::EventProcessor::init(std::shared_ptr<edm::ProcessDesc>&, edm::ServiceToken const&, edm::serviceregistry::ServiceLegacy) (/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_3_ASAN_X_2022-02-28-1100/lib/slc7_amd64_gcc11/libFWCoreFramework.so+0x2ac4de)
    #17 0x7fcb2c82f8e0 in edm::EventProcessor::EventProcessor(std::shared_ptr<edm::ProcessDesc>, edm::ServiceToken const&, edm::serviceregistry::ServiceLegacy) (/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_3_ASAN_X_2022-02-28-1100/lib/slc7_amd64_gcc11/libFWCoreFramework.so+0x2b58e0)
    #18 0x40e36b in tbb::detail::d1::task_arena_function<main::{lambda()#1}::operator()() const::{lambda()#1}, void>::operator()() const (/cvmfs/cms-ib.cern.ch/nweek-02722/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_3_ASAN_X_2022-02-28-1100/bin/slc7_amd64_gcc11/cmsRun+0x40e36b)
    #19 0x7fcb2a756897 in tbb::detail::r1::task_arena_impl::execute(tbb::detail::d1::task_arena_base&, tbb::detail::d1::delegate_base&) /data/cmsbld/jenkins/workspace/build-any-ib/w/BUILD/slc7_amd64_gcc11/external/tbb/v2021.4.0-3d940fce09ce7cacf58c68ba3ce04951/tbb-v2021.4.0/src/tbb/arena.cpp:698
    #20 0x411e7f in main::{lambda()#1}::operator()() const (/cvmfs/cms-ib.cern.ch/nweek-02722/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_3_ASAN_X_2022-02-28-1100/bin/slc7_amd64_gcc11/cmsRun+0x411e7f)
    #21 0x40c615 in main (/cvmfs/cms-ib.cern.ch/nweek-02722/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_3_ASAN_X_2022-02-28-1100/bin/slc7_amd64_gcc11/cmsRun+0x40c615)
    #22 0x7fcb29570554 in __libc_start_main (/lib64/libc.so.6+0x22554)

==3127==HINT: if you don't care about these errors you may set allocator_may_return_null=1
SUMMARY: AddressSanitizer: invalid-aligned-alloc-alignment ../../../../libsanitizer/asan/asan_malloc_linux.cpp:192 in __interceptor_aligned_alloc
==3127==ABORTING

```

**Comment by smuzaffar** [comment]
assign reconstructions

**Comment by iarspider** [comment]
@smuzaffar https://github.com/cms-sw/cmssw/issues/37017#issuecomment-1047226674

**Comment by smuzaffar** [comment]
right but ASAN issue is still open , so we need a fix for it.

**Comment by slava77** [comment]
> right but ASAN issue is still open , so we need a fix for it.

@osschar 
please clarify on the status/progress of the fix.
Thank you.

**Comment by osschar** [comment]
i'll work on it today.

**Comment by perrotta** [comment]
This should have been fixed by #37111, merged in 12_3_X since [CMSSW_12_3_ASAN_X_2022-03-04-1100](https://github.com/cms-sw/cmssw/releases/tag/CMSSW_12_3_ASAN_X_2022-03-04-1100) and [CMSSW_12_3_X_2022-03-02-2300](https://github.com/cms-sw/cmssw/releases/tag/CMSSW_12_3_X_2022-03-02-2300)

Can this issue be closed, if so?

**Comment by clacaputo** [comment]
> This should have been fixed by #37111, merged in 12_3_X since [CMSSW_12_3_ASAN_X_2022-03-04-1100](https://github.com/cms-sw/cmssw/releases/tag/CMSSW_12_3_ASAN_X_2022-03-04-1100) and [CMSSW_12_3_X_2022-03-02-2300](https://github.com/cms-sw/cmssw/releases/tag/CMSSW_12_3_X_2022-03-02-2300)
> 
> Can this issue be closed, if so?
Hi @perrotta , you're right

**Comment by clacaputo** [comment]
+reconstruction

- fixed by https://github.com/cms-sw/cmssw/pull/37111
